lib/web/device_trust.go

```diff
@@ -76,7 +81,39 @@ func (h *Handler) deviceWebConfirm(w http.ResponseWriter, r *http.Request, _ htt
 
 	// Always redirect back to the dashboard, regardless of outcome.
 	app.SetRedirectPageHeaders(w.Header(), "" /* nonce */)
-	http.Redirect(w, r, "/web", http.StatusSeeOther)
+
+	redirectTo, err := h.getRedirectURL(redirectURI)
+	if err != nil {
+		h.log.Debugf("unable to parse redirectURI %s", redirectURI)
+	}
+	http.Redirect(w, r, redirectTo, http.StatusSeeOther)
 
 	return nil, nil
 }
+
+// getRedirectURL tries to parse the given redirectURI. It will always return a redirect url
+// even if the parse fails (in case of failture, the returned string is "/web")
+func (h *Handler) getRedirectURL(redirectURI string) (string, error) {
+	if redirectURI == "" {
```

Comment:
Suggestion:

```suggestion
	const basePath = "/web"
	if redirectURI == "" {
```

and then use it throughout?